File: Accupay.Data/Services/PayPeriodStatusData.cs
using AccuPay.Data.Entities;
using AccuPay.Data.Enums;
using System.Collections.Generic;
using System.Linq;

namespace AccuPay.Data.Services
{
    public class PayPeriodStatusData
    {
        private readonly PayrollContext context;

        public enum PayPeriodStatus
        {
            Open,
            Processing,
            Closed
        }

        public class PayPeriodStatusReference
        {
            public PayPeriodStatus PayPeriodStatus { get; set; }

            public PayPeriodStatusReference(PayPeriodStatus payPeriodStatus)
            {
                this.PayPeriodStatus = payPeriodStatus;
            }
        }

        public int Index { get; set; }
        public PayPeriodStatus Status { get; set; }

        public PayPeriodStatusData(PayrollContext context)
        {
            this.context = context;
        }
    }
}

File: AccuPay/Domain/PayPeriod/PayPeriodStatusData.vb
Public Class PayPeriodStatusData

    Public Enum PayPeriodStatus
        Open
        Processing
        Closed
    End Enum

    Public Class PayPeriodStatusReference
        Public Property PayPeriodStatus As PayPeriodStatus

        Public Sub New(ByVal payPeriodStatus As PayPeriodStatus)
            Me.PayPeriodStatus = payPeriodStatus
        End Sub

    End Class

    Public Property Index As Integer
    Public Property Status As PayPeriodStatus

    Public Sub New()
    End Sub

End Class